whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -321,6 +289,22 @@ private static Autnum createAutnumResponse(final RpslObject rpslObject) {
         return autnum;
     }
 
+    private static Autnum createAsBlockResponse(final RpslObject rpslObject) {
+        final Autnum autnum = new Autnum();
+
+        String key = rpslObject.getValueForAttribute(AttributeType.AS_BLOCK).toString();
+        String[] startAndEndKey = key.split("-");
```

Comment:
Try using AsBlockRange class instead of parsing yourself (and make sure to catch and handle any exception).